backend/controllers/auth.js

```diff
@@ -34,30 +44,32 @@ const login = async (req, res, next) => {
     if (!user) {
       throw HttpError(401, { message: "Email or password is wrong" });
     }
+    const payload = {
+      id: user.id,
+    };
+    const token = jwt.sign(payload, JWT_SECRET, { expiresIn: "23h" });
+    const userWithToken = await prisma.user.update({ where: { username }, data: { token } });
 
-    return res
-      .cookie("user", JSON.stringify({ username: user.username, id: user.id }))
-      .status(200)
-      .send({
-        data: {
-          message: "Logged in succesfully",
-        },
-      });
+    return res.status(200).send({
+      data: {
+        user: { username: userWithToken.username, id: userWithToken.id, token: userWithToken.token },
+        message: "Logged in succesfully",
+      },
+    });
```

Comment:
I see multiple occurances of 
```js
res.send({
   data: {/* ... */}
})
```
This approach has a major problem, that is, that when the API changes (property names in response get changed, or new ones are added), the code has to be updated in all of the codebase. Additionaly it's not exactly clear how other endpoints should create their API (should they use the same structure or come up with a different way of sending data?). This leads to unmaintainable code. 

I suggest you create a separate `createApiResponse()` function somewhere, and use that to generate API responses. Example:
```js
function createApiResponse(error, payload) {
 return {
  success: Boolean(error),
  data: payload,
 }
}
```
Additional success variable makes it easier to handle errors on the frontend